stdhttp: mixed path parameter and literal suffix generates invalid ServeMux pattern

## Version

- `oapi-codegen`: `v2.8.0`
- Go: `go1.26.5`
- Generator: `std-http-server`

## Minimal reproduction

Configuration:

```yaml
package: api
output: server.gen.go
generate:
  std-http-server: true
```

OpenAPI specification:

```yaml
openapi: 3.0.3
info:
  title: Custom action reproduction
  version: 1.0.0
paths:
  /resources/{resourceId}:apply:
    post:
      operationId: applyResource
      parameters:
        - name: resourceId
          in: path
          required: true
          schema:
            type: string
      responses:
        "204":
          description: Applied
```

The generated `std-http-server` registration is equivalent to:

```go
m.HandleFunc(
	http.MethodPost+" "+options.BaseURL+"/resources/{resourceId}:apply",
	wrapper.ApplyResource,
)
```

Registering that pattern with `http.ServeMux` panics:

```text
panic: parsing "POST /resources/{resourceId}:apply": at offset 16: bad wildcard segment (must end with '}')
```

This can also be reproduced without generated code:

```go
mux := http.NewServeMux()
mux.HandleFunc(
	"POST /resources/{resourceId}:apply",
	func(http.ResponseWriter, *http.Request) {},
)
```

Although Go 1.22+ supports whole-segment wildcards, a mixed wildcard and literal
segment is not valid. Registering `/resources/{resourceId}` instead accepts the
request, but `PathValue("resourceId")` returns `abc:apply` for
`/resources/abc:apply`; `ServeMux` does not strip the action suffix.

## Expected behavior

For `std-http-server`, either:

1. generate a compatible route that matches the whole segment, validates and
   strips the literal suffix, and passes `abc` as `resourceId`; or
2. reject this path shape during generation with a clear unsupported-route
   error instead of generating code that panics during server startup.

Literal suffixes on templated path segments are useful for custom action routes
such as `{resourceId}:apply`.

## Additional context

The v2.8.0 trailing-slash and handler-registration-order fixes do not cover this
path shape. A local workaround is to register
`/resources/{resourceIdAction}`, dispatch based on the `:apply` suffix, and set
the original `resourceId` path value before invoking the generated wrapper.

The minimal reproduction snippets in this issue may be used under Apache-2.0.

--- a/pkg/codegen/codegen.go
+++ b/pkg/codegen/codegen.go
@@ -178,6 +178,11 @@ func Generate(spec *openapi3.T, opts Configuration) (string, error) {
 	if err := ValidateSpec(spec); err != nil {
 		return "", err
 	}
+	if opts.Generate.StdHTTPServer {
+		if err := ValidateStdHTTPPaths(spec); err != nil {
+			return "", err
+		}
+	}
 
 	// if we are provided an override for the response type suffix update it
 	if opts.OutputOptions.ResponseTypeSuffix != "" {

--- a/pkg/codegen/utils.go
+++ b/pkg/codegen/utils.go
@@ -15,6 +15,7 @@ package codegen
 
 import (
 	"bytes"
+	"errors"
 	"cmp"
 	"fmt"
 	"go/token"
@@ -680,6 +681,42 @@ func SwaggerUriToStdHttpUri(uri string) string {
 	}
 
 	return uri
+}
+
+// ValidateStdHTTPPath reports whether an OpenAPI path can be registered with
+// net/http ServeMux. ServeMux wildcards must occupy an entire path segment, so
+// mixed segments such as "{resourceId}:apply" are rejected at codegen time
+// instead of panicking at server startup (see #2488).
+func ValidateStdHTTPPath(path string) error {
+	for _, seg := range strings.Split(path, "/") {
+		if seg == "" || seg == "{$}" {
+			continue
+		}
+		loc := pathParamRE.FindStringIndex(seg)
+		if loc == nil {
+			// pure literal segment
+			continue
+		}
+		// Wildcard must be the entire segment.
+		if loc[0] != 0 || loc[1] != len(seg) {
+			return fmt.Errorf("path %q: segment %q mixes a path parameter with literal text; net/http ServeMux requires wildcards to occupy an entire path segment (std-http-server)", path, seg)
+		}
+	}
+	return nil
+}
+
+// ValidateStdHTTPPaths validates every path in the document for ServeMux.
+func ValidateStdHTTPPaths(spec *openapi3.T) error {
+	if spec == nil || spec.Paths == nil {
+		return nil
+	}
+	var errs []error
+	for _, path := range SortedMapKeys(spec.Paths.Map()) {
+		if err := ValidateStdHTTPPath(path); err != nil {
+			errs = append(errs, err)
+		}
+	}
+	return errors.Join(errs...)
 }
 
 // OrderedParamsFromUri returns the argument names, in order, in a given URI string, so for

--- a/pkg/codegen/utils.go
+++ b/pkg/codegen/utils.go
@@ -706,12 +706,19 @@ func ValidateStdHTTPPath(path string) error {
 }
 
 // ValidateStdHTTPPaths validates every path in the document for ServeMux.
+// Paths whose operations have all been removed (e.g. by tag or operation-id
+// filtering) emit no routes, so they are skipped.
 func ValidateStdHTTPPaths(spec *openapi3.T) error {
 	if spec == nil || spec.Paths == nil {
 		return nil
 	}
 	var errs []error
-	for _, path := range SortedMapKeys(spec.Paths.Map()) {
+	pathMap := spec.Paths.Map()
+	for _, path := range SortedMapKeys(pathMap) {
+		pathItem := pathMap[path]
+		if pathItem == nil || len(pathItem.Operations()) == 0 {
+			continue
+		}
 		if err := ValidateStdHTTPPath(path); err != nil {
 			errs = append(errs, err)
 		}

--- a/pkg/codegen/utils_test.go
+++ b/pkg/codegen/utils_test.go
@@ -837,3 +837,62 @@ func Test_replaceInitialism(t *testing.T) {
 		})
 	}
 }
+
+
+func TestValidateStdHTTPPath(t *testing.T) {
+	tests := []struct {
+		path    string
+		wantErr bool
+	}{
+		{"/resources/{resourceId}", false},
+		{"/resources/{resourceId}/apply", false},
+		{"/resources/{resourceId}:apply", true},
+		{"/resources/prefix{resourceId}", true},
+		{"/resources/{resourceId}suffix", true},
+		{"/pets:validate", false}, // pure literal segment with colon is fine for ServeMux
+		{"/path/{arg1}/{arg2}/foo", false},
+	}
+	for _, tt := range tests {
+		err := ValidateStdHTTPPath(tt.path)
+		if tt.wantErr && err == nil {
+			t.Errorf("path %q: expected error", tt.path)
+		}
+		if !tt.wantErr && err != nil {
+			t.Errorf("path %q: unexpected error: %v", tt.path, err)
+		}
+	}
+}
+
+func TestGenerateRejectsMixedServeMuxPathParam(t *testing.T) {
+	spec := `openapi: 3.0.3
+info:
+  title: mixed path param
+  version: 1.0.0
+paths:
+  /resources/{resourceId}:apply:
+    post:
+      operationId: applyResource
+      parameters:
+        - name: resourceId
+          in: path
+          required: true
+          schema:
+            type: string
+      responses:
+        "204":
+          description: Applied
+`
+	loader := openapi3.NewLoader()
+	swagger, err := loader.LoadFromData([]byte(spec))
+	require.NoError(t, err)
+
+	_, err = Generate(swagger, Configuration{
+		PackageName: "api",
+		Generate: GenerateOptions{
+			StdHTTPServer: true,
+			Models:        true,
+		},
+	})
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "mixes a path parameter")
+}

--- a/pkg/codegen/utils_test.go
+++ b/pkg/codegen/utils_test.go
@@ -896,3 +896,49 @@ paths:
 	require.Error(t, err)
 	require.Contains(t, err.Error(), "mixes a path parameter")
 }
+
+func TestGenerateAllowsMixedServeMuxPathParamWhenFilteredOut(t *testing.T) {
+	spec := `openapi: 3.0.3
+info:
+  title: mixed path param filtered out
+  version: 1.0.0
+paths:
+  /resources/{resourceId}:apply:
+    post:
+      operationId: applyResource
+      tags: [excluded]
+      parameters:
+        - name: resourceId
+          in: path
+          required: true
+          schema:
+            type: string
+      responses:
+        "204":
+          description: Applied
+  /pets:
+    get:
+      operationId: listPets
+      responses:
+        "200":
+          description: OK
+`
+	loader := openapi3.NewLoader()
+	swagger, err := loader.LoadFromData([]byte(spec))
+	require.NoError(t, err)
+
+	// Tag filtering removes the only operation on the mixed segment but leaves
+	// the path key in the spec; no route is emitted for it, so generation must
+	// succeed.
+	_, err = Generate(swagger, Configuration{
+		PackageName: "api",
+		Generate: GenerateOptions{
+			StdHTTPServer: true,
+			Models:        true,
+		},
+		OutputOptions: OutputOptions{
+			ExcludeTags: []string{"excluded"},
+		},
+	})
+	require.NoError(t, err)
+}

